### PR TITLE
Feature/radio dialog

### PR DESCRIPTION
**Pull request by BartSoftware**

### CODE REVIEW

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/common/RadioDialog.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package com.invisibleteam.goinvisible.mvvm.common;
+
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+
+public class RadioDialog {
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/common/RadioDialog.java]: I would change name of this class to RadioDialogBuilder

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/dialog/DialogFactory.java [hunk 2]
```diff
@@ -79,33 +78,57 @@ Dialog createDialog() {
     }
 
     @Nullable
-    Dialog createRangedDialog(Context context, Tag tag, EditViewModel viewModel) {
-        AlertDialog.Builder alertDialog = new AlertDialog.Builder(context);
-        alertDialog.setTitle(tag.getKey());
+    Dialog createRangedDialog(Context context, Tag baseTag, EditViewModel viewModel) {
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/dialog/DialogFactory.java]: I think there is too much going on in this method. I would like you to split this method in multiple methods, or extract some code to separate classes.

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/settings/SettingsActivity.java [hunk 3]
```diff
@@ -95,23 +96,43 @@ private void openIntervalsDialog() {
         alertDialog.show();
     }
 
-    private AlertDialog createIntervalsDialog() {
-        AlertDialog.Builder alertDialog = new AlertDialog.Builder(this);
-        alertDialog.setTitle(getString(R.string.intervals_dialog_title));
-
+    private Dialog createIntervalsDialog() {
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/settings/SettingsActivity.java:99]: I think there is too much going on in this method. I would like you to split this method in multiple methods, or extract some code to separate classes.

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/dialog/DialogFactory.java [hunk 4]
```diff
@@ -79,33 +78,61 @@ Dialog createDialog() {
...
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/dialog/DialogFactory.java]: EditViewModel viewModel is not used in this method.

**Review by kejmil01 — APPROVED** [review]

**Review by michal1990 — APPROVED** [review]

**Pull request merged**